messaging/src/main/java/org/axonframework/messaging/ClassBasedMessageTypeResolver.java

```diff
@@ -57,8 +56,8 @@ public ClassBasedMessageTypeResolver(String version) {
     }
 
     @Override
-    public MessageType resolve(@Nonnull Class<?> payloadType) {
+    public Optional<MessageType> resolve(Class<?> payloadType) {
```

Comment:
Why was the `@Nonnull` dropped here? :thinking:

Reply:
Because it's not in the interface, but I think it should be - I will add it :D